xload/providers/cached/loader.go

```diff
@@ -0,0 +1,37 @@
+package cached
+
+import (
+	"context"
+
+	"github.com/gojekfarm/xtools/xload"
+)
+
+// NewLoader returns a new cached loader.
+func NewLoader(l xload.Loader, opts ...Option) xload.LoaderFunc {
+	o := defaultOptions()
+
+	o.apply(opts...)
+
+	return xload.LoaderFunc(func(ctx context.Context, key string) (string, error) {
+		v, err := o.cache.Get(key)
+		if err != nil {
+			return "", err
+		}
+
+		if v != "" {
+			return v, nil
+		}
+
+		loaded, err := l.Load(ctx, key)
+		if err != nil {
+			return "", err
+		}
+
+		// DESIGN: If the loader returns an empty value, we
```

Comment:
Can we make this configurable via an `Option` say `DisableEmptyValueHit`. Default can remain the same as now.

```go
var DisableEmptyValueHit = emptyValHit(false)

type emptyValHit bool
func (e emptyValHit) apply(o *options) { o.emptyHit = bool(e) }
```

Reply:
Added a functional option